File: mobile/lib/screens/auth/cloudflare_login_screen.dart
// ═══════════════════════════════════════════════════════════════
//  cloudflare_login_screen.dart — شاشة تسجيل الدخول إلى Cloudflare
//  ✅ (2026-09-10) طلب المستخدم: «اضف شاشة تسجيل الدخول الى cloudflare»
//
//  ماذا تعرض:
//   • حالة الاتصال الحيّة (من syncStatusStream) + حالة تسجيل الدخول
//   • نقطة النهاية الفعّالة (النطاق المخصّص أو workers.dev)
//   • معرّف الجهاز واسم المستخدم الحالي
//   • رسالة آخر خطأ تهيئة (initError) إن وُجدت — بنص عربي مفهوم
//
//  ماذا تفعل:
//   • تُدخل اسم مستخدم/كلمة مرور بديلة (overrides) تُحفظ محلياً
//     وتعمل على حساب المدمج --dart-define — فيصلح «لم يتم تسجيل
//     الدخول إلى سيرفر المزامنة» دون إعادة بناء APK
//   • زر «تسجيل الدخول»: يحفظ الاعتمادات ثم initialize(forceRetry)
//   • زر «فحص الاتصال»: يصيب /health ويعرض النتيجة
//   • زر «الرجوع للاعتمادات المدمجة»: يمسح overrides
// ═══════════════════════════════════════════════════════════════

import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';

import '../../providers/appwrite_providers.dart';
import '../../services/appwrite_sync_manager.dart' show SyncStatus;
import '../../services/cloudflare_config.dart';
import '../../services/worker_endpoints.dart';
import '../../utils/theme.dart';

class CloudflareLoginScreen extends ConsumerStatefulWidget {
  const CloudflareLoginScreen({super.key});

  @override
  ConsumerState<CloudflareLoginScreen> createState() =>
      _CloudflareLoginScreenState();
}

class _CloudflareLoginScreenState extends ConsumerState<CloudflareLoginScreen> {
  final _usernameController = TextEditingController();
  final _passwordController = TextEditingController();
  bool _isLoggingIn = false;
  bool _isCheckingHealth = false;
  bool _obscurePassword = true;
  String? _message;
  Color? _messageColor;
  String? _healthResult;

  @override
  void initState() {
    super.initState();
    // نملأ اسم المستخدم بالقيمة الفعّالة (override أو مدمج) ليعدّل
    // فوقها بدل الكتابة من الصفر. كلمة المرور لا تُعاد أبداً — الحقل
    // فارغ = «إبقاء الحالية» كما هو موثّق أسفل الحقل.
    _usernameController.text = CloudflareConfig.username;
  }

  @override
  void dispose() {
    _usernameController.dispose();
    _passwordController.dispose();
    super.dispose();
  }

  Future<void> _login() async {
    setState(() {
      _isLoggingIn = true;
      _message = null;
      _healthResult = null;
    });
    try {
      // 1) حفظ الاعتمادات (فارغ = إبقاء) — تعمل فوراً على getters
      await CloudflareConfig.setCredentialOverrides(
        username: _usernameController.text,
        password: _passwordController.text.isEmpty
            ? null
            : _passwordController.text,
      );

      // 2) تهيئة إجبارية — تُعيد تسجيل الدخول بالاعتمادات الجديدة
      final manager = ref.read(appwriteSyncManagerProvider);
      await manager.initialize(forceRetry: true, loginAttempts: 2);

      if (!mounted) return;
      setState(() {
        _isLoggingIn = false;
        if (manager.isAvailable) {
          _message = '✅ تم تسجيل الدخول بنجاح — المزامنة جاهزة';
          _messageColor = AppColors.successColor;
          _passwordController.clear();
        } else {
          _message = manager.initError ?? 'فشل تسجيل الدخول — راجع البيانات';
          _messageColor = AppColors.dangerColor;
        }
      });
    } catch (e) {
      if (!mounted) {
        return;
      }
      setState(() {
        _isLoggingIn = false;
        _message = 'خطأ: $e';
        _messageColor = AppColors.dangerColor;
      });
    }
  }

  Future<void> _checkHealth() async {
    setState(() {
      _isCheckingHealth = true;
      _healthResult = null;
    });
    final notifier = ref.read(connectionStatusProvider.notifier);
    await notifier.checkConnection();
    if (!mounted) return;
    final state = ref.read(connectionStatusProvider);
    setState(() {
      _isCheckingHealth = false;
      _healthResult = state.isConnected
          ? '✅ الاتصال بخادم المزامنة يعمل (${WorkerEndpoints.active})'
          : '❌ تعذر الوصول للخادم: ${state.errorMessage ?? 'غير معروف'}';
    });
  }

  Future<void> _resetOverrides() async {
    await CloudflareConfig.clearCredentialOverrides();
    if (!mounted) return;
    setState(() {
      _usernameController.text = CloudflareConfig.username;
      _passwordController.clear();
      _message = 'أُزيلت الاعتمادات المخصّصة — الرجوع للمدمجة';
      _messageColor = AppColors.infoColor;
    });
  }

  @override
  Widget build(BuildContext context) {
    final statusAsync = ref.watch(cloudflareSyncStatusProvider);
    final status = statusAsync.when(
      data: (s) => s,
      loading: () => SyncStatus.idle,
      error: (_, __) => SyncStatus.failed,
    );
    final workerUrl = WorkerEndpoints.active;

    return Directionality(
      textDirection: TextDirection.rtl,
      child: Scaffold(
        backgroundColor: AppColors.backgroundColor,
        appBar: AppBar(
          title: const Text('تسجيل الدخول إلى Cloudflare'),
          backgroundColor: AppColors.primaryColor,
          foregroundColor: Colors.white,
        ),
        body: SingleChildScrollView(
          padding: const EdgeInsets.all(16),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.stretch,
            children: [
              // ─── بطاقة الحالة الحالية ───
              _StatusCard(status: status, workerUrl: workerUrl),
              const SizedBox(height: 16),

              // ─── رسالة آخر خطأ تهيئة إن وُجد ───
              _buildInitErrorBanner(),

              // ─── نموذج الاعتمادات ───
              Card(
                child: Padding(
                  padding: const EdgeInsets.all(16),
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.stretch,
                    children: [
                      const Row(
                        children: [
                          Icon(Icons.key, color: AppColors.primaryColor),
                          SizedBox(width: 8),
                          Text(
                            'اعتمادات حساب المزامنة',
                            style: TextStyle(
                              fontSize: 16,
                              fontWeight: FontWeight.bold,
                            ),
                          ),
                        ],
                      ),
                      const SizedBox(height: 4),
                      const Text(
                        'اترك كلمة المرور فارغة للإبقاء على الحالية.',
                        style: TextStyle(
                          fontSize: 12,
                          color: AppColors.textSecondary,
                        ),
                      ),
                      const SizedBox(height: 12),
                      TextField(
                        controller: _usernameController,
                        decoration: const InputDecoration(
                          labelText: 'اسم المستخدم',
                          prefixIcon: Icon(Icons.person_outline),
                          border: OutlineInputBorder(),
                        ),
                        enabled: !_isLoggingIn,
                      ),
                      const SizedBox(height: 12),
                      TextField(
                        controller: _passwordController,
                        obscureText: _obscurePassword,
                        decoration: InputDecoration(
                          labelText: 'كلمة المرور',
                          prefixIcon: const Icon(Icons.lock_outline),
                          border: const OutlineInputBorder(),
                          suffixIcon: IconButton(
                            icon: Icon(
                              _obscurePassword
                                  ? Icons.visibility
                                  : Icons.visibility_off,
                            ),
                            onPressed: () => setState(
                              () => _obscurePassword = !_obscurePassword,
                            ),
                          ),
                        ),
                        enabled: !_isLoggingIn,
                      ),
                      const SizedBox(height: 16),
                      ElevatedButton.icon(
                        onPressed: _isLoggingIn ? null : _login,
                        icon: _isLoggingIn
                            ? const SizedBox(
                                width: 18,
                                height: 18,
                                child: CircularProgressIndicator(
                                  strokeWidth: 2,
                                  color: Colors.white,
                                ),
                              )
                            : const Icon(Icons.login),
                        label: Text(
                          _isLoggingIn
                              ? 'جارٍ تسجيل الدخول...'
                              : 'تسجيل الدخول',
                        ),
                        style: ElevatedButton.styleFrom(
                          padding: const EdgeInsets.symmetric(vertical: 14),
                          backgroundColor: AppColors.primaryColor,
                          foregroundColor: Colors.white,
                        ),
                      ),
                      const SizedBox(height: 8),
                      Row(
                        children: [
                          Expanded(
                            child: OutlinedButton.icon(
                              onPressed: _isCheckingHealth
                                  ? null
                                  : _checkHealth,
                              icon: _isCheckingHealth
                                  ? const SizedBox(
                                      width: 16,
                                      height: 16,
                                      child: CircularProgressIndicator(
                                        strokeWidth: 2,
                                      ),
                                    )
                                  : const Icon(Icons.wifi_tethering),
                              label: const Text('فحص الاتصال'),
                            ),
                          ),
                          if (CloudflareConfig.hasCredentialOverrides) ...[
                            const SizedBox(width: 8),
                            TextButton.icon(
                              onPressed: (_isLoggingIn || _isCheckingHealth)
                                  ? null
                                  : _resetOverrides,
                              icon: const Icon(Icons.restart_alt, size: 18),
                              label: const Text('الاعتمادات المدمجة'),
                            ),
                          ],
                        ],
                      ),
                    ],
                  ),
                ),
              ),
              if (_message != null) ...[
                const SizedBox(height: 12),
                _buildBanner(_message!, _messageColor ?? AppColors.infoColor),
              ],
              if (_healthResult != null) ...[
                const SizedBox(height: 8),
                _buildBanner(
                  _healthResult!,
                  ref.read(connectionStatusProvider).isConnected
                      ? AppColors.successColor
                      : AppColors.dangerColor,
                ),
              ],
              const SizedBox(height: 16),

              // ─── شرح مختصر ───
              Container(
                padding: const EdgeInsets.all(12),
                decoration: BoxDecoration(
                  color: AppColors.infoColor.withValues(alpha: 0.08),
                  borderRadius: BorderRadius.circular(8),
                ),
                child: const Text(
                  'تُحفظ الاعتمادات على هذا الجهاز فقط وتُستخدم مع كل عمليات '
                  'المزامنة (دفع/سحب). إذا كان workers.dev محجوباً في شبكتك '
                  'أضف نطاقاً مخصّصاً من إعدادات المزامنة أولاً.',
                  style: TextStyle(fontSize: 12, height: 1.6),
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }

  Widget _buildInitErrorBanner() {
    final manager = ref.watch(appwriteSyncManagerProvider);
    final initError = manager.initError;
    if (initError == null || initError.isEmpty) return const SizedBox.shrink();
    return Padding(
      padding: const EdgeInsets.only(bottom: 16),
      child: _buildBanner(initError, AppColors.dangerColor),
    );
  }

  Widget _buildBanner(String text, Color color) {
    return Container(
      padding: const EdgeInsets.all(12),
      decoration: BoxDecoration(
        color: color.withValues(alpha: 0.1),
        borderRadius: BorderRadius.circular(8),
        border: Border.all(color: color.withValues(alpha: 0.4)),
      ),
      child: Row(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Icon(Icons.info_outline, color: color, size: 20),
          const SizedBox(width: 8),
          Expanded(
            child: Text(
              text,
              style: TextStyle(color: color, fontSize: 13, height: 1.5),
            ),
          ),
        ],
      ),
    );
  }
}

/// بطاقة الحالة العلوية — لحظية عبر cloudflareSyncStatusProvider.
class _StatusCard extends StatelessWidget {
  const _StatusCard({required this.status, required this.workerUrl});
  final SyncStatus status;
  final String workerUrl;

  @override
  Widget build(BuildContext context) {
    final (icon, color, label) = switch (status) {
      SyncStatus.syncing => (
        Icons.sync,
        Colors.blue,
        'جاري المزامنة الآن...',
      ),
      SyncStatus.success => (
        Icons.cloud_done,
        Colors.green,
        'متصل — آخر مزامنة نجحت',
      ),
      SyncStatus.failed => (
        Icons.cloud_off,
        Colors.orange,
        'آخر مزامنة فشلت — جرّب تسجيل الدخول أدناه',
      ),
      _ => (Icons.cloud_outlined, Colors.grey, 'جاهز — لا مزامنة جارية'),
    };

    return Card(
      child: Padding(
        padding: const EdgeInsets.all(16),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Row(
              children: [
                Icon(icon, color: color, size: 28),
                const SizedBox(width: 12),
                Expanded(
                  child: Text(
                    label,
                    style: const TextStyle(
                      fontSize: 15,
                      fontWeight: FontWeight.bold,
                    ),
                  ),
                ),
              ],
            ),
            const SizedBox(height: 12),
            _row(Icons.dns, 'الخادم', workerUrl),
            const SizedBox(height: 6),
            _row(Icons.key, 'الحساب', CloudflareConfig.username),
          ],
        ),
      ),
    );
  }

  Widget _row(IconData icon, String label, String value) {
    return Row(
      children: [
        Icon(icon, size: 16, color: AppColors.textSecondary),
        const SizedBox(width: 8),
        Text('$label: ', style: const TextStyle(fontSize: 13)),
        Expanded(
          child: Text(
            value,
            style: const TextStyle(
              fontSize: 13,
              fontWeight: FontWeight.w600,
              fontFamily: 'monospace',
            ),
            overflow: TextOverflow.ellipsis,
          ),
        ),
      ],
    );
  }
}

File: mobile/android/app/src/main/kotlin/com/marina/marina/presentation/auth/CloudflareLoginScreen.kt
package com.marina.marina.presentation.auth

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.CloudDone
import androidx.compose.material.icons.filled.CloudOff
import androidx.compose.material.icons.filled.Dns
import androidx.compose.material.icons.filled.Info
import androidx.compose.material.icons.filled.Key
import androidx.compose.material.icons.filled.Login
import androidx.compose.material.icons.filled.PersonOutline
import androidx.compose.material.icons.filled.RestartAlt
import androidx.compose.material.icons.filled.Sync
import androidx.compose.material.icons.filled.Visibility
import androidx.compose.material.icons.filled.VisibilityOff
import androidx.compose.material.icons.filled.WifiTethering
import androidx.compose.material.icons.outlined.Cloud
import androidx.compose.material.icons.outlined.Lock
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.material3.TopAppBar
import androidx.compose.material3.TopAppBarDefaults
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.text.input.PasswordVisualTransformation
import androidx.compose.ui.text.input.VisualTransformation
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.hilt.navigation.compose.hiltViewModel
import com.marina.marina.presentation.auth.CloudflareConnectionViewModel.CfSyncStatus
import com.marina.marina.presentation.auth.CloudflareConnectionViewModel.MessageKind
import com.marina.marina.ui.theme.AppColors

/**
 * ✅ (2026-09-24) شاشة «تسجيل الدخول إلى Cloudflare» — نقل 1:1 لـ
 * cloudflare_login_screen.dart (فرع feat/cloudflare-sync-execution):
 *
 *  • بطاقة الحالة العلوية: أيقونة/لون/نص حسب حالة المزامنة + صفّا
 *    «الخادم» (النقطة الفعّالة) و«الحساب» (بخط monospace).
 *  • بانر خطأ التهيئة (initError) إن وُجد.
 *  • بطاقة الاعتمادات: عنوان بأيقونة مفتاح + ملاحظة الإبقاء على كلمة
 *    المرور + حقلان مع أيقونات بادئة + زر دخول بمؤشر تحميل + فحص
 *    الاتصال + زر الاعتمادات المدمجة (عند وجود overrides).
 *  • بانرا الرسالة وفحص الصحة + حاوية الشرح المختصر.
 */
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun CloudflareLoginScreen(
    onBack: () -> Unit = {},
    viewModel: CloudflareConnectionViewModel = hiltViewModel()
) {
    val state by viewModel.state.collectAsState()

    // دخول تلقائي واحد عند فتح الشاشة (نظير addPostFrameCallback في Dart).
    LaunchedEffect(Unit) { viewModel.autoLoginIfNeeded() }

    Scaffold(
        containerColor = AppColors.BackgroundColor,
        topBar = {
            TopAppBar(
                title = { Text("تسجيل الدخول إلى Cloudflare") },
                navigationIcon = {
                    IconButton(onClick = onBack) {
                        Icon(Icons.AutoMirrored.Filled.ArrowBack, contentDescription = "رجوع")
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = AppColors.PrimaryColor,
                    titleContentColor = Color.White,
                    navigationIconContentColor = Color.White
                )
            )
        }
    ) { padding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(padding)
                .verticalScroll(rememberScrollState())
                .padding(16.dp)
        ) {
            // ─── بطاقة الحالة الحالية (_StatusCard) ───
            StatusCard(
                status = state.status,
                workerUrl = state.workerUrl,
                account = state.account
            )
            Spacer(Modifier.height(16.dp))

            // ─── بانر آخر خطأ تهيئة (_buildInitErrorBanner) ───
            state.initError?.let { initError ->
                Banner(text = initError, color = AppColors.DangerColor)
                Spacer(Modifier.height(16.dp))
            }

            // ─── بطاقة الاعتمادات ───
            Card(
                shape = RoundedCornerShape(12.dp),
                colors = CardDefaults.cardColors(containerColor = AppColors.SurfaceColor),
                elevation = CardDefaults.cardElevation(defaultElevation = 1.dp),
                modifier = Modifier.fillMaxWidth()
            ) {
                Column(modifier = Modifier.padding(16.dp)) {
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Icon(Icons.Default.Key, contentDescription = null, tint = AppColors.PrimaryColor, modifier = Modifier.size(20.dp))
                        Spacer(Modifier.width(8.dp))
                        Text("اعتمادات حساب المزامنة", fontSize = 16.sp, fontWeight = FontWeight.Bold)
                    }
                    Spacer(Modifier.height(4.dp))
                    Text(
                        "اترك كلمة المرور فارغة للإبقاء على الحالية.",
                        fontSize = 12.sp,
                        color = AppColors.TextSecondary
                    )
                    Spacer(Modifier.height(12.dp))
                    OutlinedTextField(
                        value = state.usernameField,
                        onValueChange = viewModel::onUsernameChange,
                        label = { Text("اسم المستخدم") },
                        leadingIcon = { Icon(Icons.Default.PersonOutline, contentDescription = null) },
                        singleLine = true,
                        enabled = !state.isLoggingIn,
                        keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Text),
                        modifier = Modifier.fillMaxWidth()
                    )
                    Spacer(Modifier.height(12.dp))
                    OutlinedTextField(
                        value = state.passwordField,
                        onValueChange = viewModel::onPasswordChange,
                        label = { Text("كلمة المرور") },
                        leadingIcon = { Icon(Icons.Outlined.Lock, contentDescription = null) },
                        singleLine = true,
                        enabled = !state.isLoggingIn,
                        visualTransformation = if (state.obscurePassword) {
                            PasswordVisualTransformation()
                        } else {
                            VisualTransformation.None
                        },
                        trailingIcon = {
                            IconButton(onClick = viewModel::toggleObscurePassword) {
                                Icon(
                                    if (state.obscurePassword) Icons.Default.Visibility else Icons.Default.VisibilityOff,
                                    contentDescription = null
                                )
                            }
                        },
                        keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Password),
                        modifier = Modifier.fillMaxWidth()
                    )
                    Spacer(Modifier.height(16.dp))
                    androidx.compose.material3.Button(
                        onClick = viewModel::login,
                        enabled = !state.isLoggingIn,
                        colors = androidx.compose.material3.ButtonDefaults.buttonColors(
                            containerColor = AppColors.PrimaryColor,
                            contentColor = Color.White
                        ),
                        modifier = Modifier.fillMaxWidth()
                    ) {
                        if (state.isLoggingIn) {
                            CircularProgressIndicator(
                                modifier = Modifier.size(18.dp),
                                strokeWidth = 2.dp,
                                color = Color.White
                            )
                            Spacer(Modifier.width(8.dp))
                            Text("جارٍ تسجيل الدخول...")
                        } else {
                            Icon(Icons.Default.Login, contentDescription = null, modifier = Modifier.size(18.dp))
                            Spacer(Modifier.width(8.dp))
                            Text("تسجيل الدخول")
                        }
                    }
                    Spacer(Modifier.height(8.dp))
                    Row(
                        verticalAlignment = Alignment.CenterVertically,
                        horizontalArrangement = Arrangement.spacedBy(8.dp)
                    ) {
                        OutlinedButton(
                            onClick = viewModel::checkHealth,
                            enabled = !state.isCheckingHealth,
                            modifier = Modifier.weight(1f)
                        ) {
                            if (state.isCheckingHealth) {
                                CircularProgressIndicator(modifier = Modifier.size(16.dp), strokeWidth = 2.dp)
                            } else {
                                Icon(Icons.Default.WifiTethering, contentDescription = null, modifier = Modifier.size(18.dp))
                                Spacer(Modifier.width(6.dp))
                                Text("فحص الاتصال")
                            }
                        }
                        if (state.hasCredentialOverrides) {
                            TextButton(
                                onClick = viewModel::resetOverrides,
                                enabled = !state.isLoggingIn && !state.isCheckingHealth
                            ) {
                                Icon(Icons.Default.RestartAlt, contentDescription = null, modifier = Modifier.size(18.dp))
                                Spacer(Modifier.width(4.dp))
                                Text("الاعتمادات المدمجة")
                            }
                        }
                    }
                }
            }

            // ─── بانر الرسالة (_message) ───
            state.message?.let { message ->
                Spacer(Modifier.height(12.dp))
                Banner(
                    text = message.text,
                    color = when (message.kind) {
                        MessageKind.SUCCESS -> AppColors.SuccessColor
                        MessageKind.DANGER -> AppColors.DangerColor
                        MessageKind.INFO -> AppColors.InfoColor
                    }
                )
            }

            // ─── بانر نتيجة فحص الاتصال (_healthResult) ───
            state.healthResult?.let { health ->
                Spacer(Modifier.height(8.dp))
                Banner(
                    text = health.text,
                    color = if (health.kind == MessageKind.SUCCESS) AppColors.SuccessColor else AppColors.DangerColor
                )
            }

            Spacer(Modifier.height(16.dp))

            // ─── شرح مختصر (Container infoColor 8% في Dart) ───
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .background(
                        color = AppColors.InfoColor.copy(alpha = 0.08f),
                        shape = RoundedCornerShape(8.dp)
                    )
                    .padding(12.dp)
            ) {
                Text(
                    "تُحفظ الاعتمادات على هذا الجهاز فقط وتُستخدم مع كل عمليات " +
                        "المزامنة (دفع/سحب). إذا كان workers.dev محجوباً في شبكتك " +
                        "أضف نطاقاً مخصّصاً من إعدادات المزامنة أولاً.",
                    fontSize = 12.sp,
                    lineHeight = 19.sp
                )
            }
        }
    }
}

// ═══════════════ مكونات مطابقة لعناصر Dart ═══════════════

/** ألوان حالات SyncStatus في Dart (Colors.blue/green/orange/grey). */
private object StatusColors {
    val blue = Color(0xFF2196F3)
    val green = Color(0xFF4CAF50)
    val orange = Color(0xFFFF9800)
    val grey = Color(0xFF9E9E9E)
}

/** بطاقة الحالة العلوية — لحظية (نظير _StatusCard في Dart). */
@Composable
private fun StatusCard(status: CfSyncStatus, workerUrl: String, account: String) {
    val (icon, color, label) = when (status) {
        CfSyncStatus.SYNCING -> Triple(Icons.Default.Sync, StatusColors.blue, "جاري المزامنة الآن...")
        CfSyncStatus.SUCCESS -> Triple(Icons.Default.CloudDone, StatusColors.green, "متصل — آخر مزامنة نجحت")
        CfSyncStatus.FAILED -> Triple(Icons.Default.CloudOff, StatusColors.orange, "آخر مزامنة فشلت — جرّب تسجيل الدخول أدناه")
        CfSyncStatus.IDLE -> Triple(Icons.Outlined.Cloud, StatusColors.grey, "جاهز — لا مزامنة جارية")
    }

    Card(
        shape = RoundedCornerShape(12.dp),
        colors = CardDefaults.cardColors(containerColor = AppColors.SurfaceColor),
        elevation = CardDefaults.cardElevation(defaultElevation = 1.dp),
        modifier = Modifier.fillMaxWidth()
    ) {
        Column(modifier = Modifier.padding(16.dp)) {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Icon(icon, contentDescription = null, tint = color, modifier = Modifier.size(28.dp))
                Spacer(Modifier.width(12.dp))
                Text(
                    label,
                    fontSize = 15.sp,
                    fontWeight = FontWeight.Bold,
                    modifier = Modifier.weight(1f)
                )
            }
            Spacer(Modifier.height(12.dp))
            StatusRow(Icons.Default.Dns, "الخادم", workerUrl)
            Spacer(Modifier.height(6.dp))
            StatusRow(Icons.Default.Key, "الحساب", account)
        }
    }
}

/** صف معلومة — أيقونة 16 + تسمية 13 + قيمة 13 w600 monospace (Dart). */
@Composable
private fun StatusRow(icon: androidx.compose.ui.graphics.vector.ImageVector, label: String, value: String) {
    Row(verticalAlignment = Alignment.CenterVertically) {
        Icon(icon, contentDescription = null, tint = AppColors.TextSecondary, modifier = Modifier.size(16.dp))
        Spacer(Modifier.width(8.dp))
        Text("$label: ", fontSize = 13.sp)
        Text(
            value,
            fontSize = 13.sp,
            fontWeight = FontWeight.SemiBold,
            fontFamily = FontFamily.Monospace,
            maxLines = 1,
            overflow = TextOverflow.Ellipsis,
            modifier = Modifier.weight(1f)
        )
    }
}

/** بانر ملوّن — خلفية 10% + حدود 40% + أيقونة معلومات (نظير _buildBanner). */
@Composable
private fun Banner(text: String, color: Color) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .background(
                color = color.copy(alpha = 0.1f),
                shape = RoundedCornerShape(8.dp)
            )
            .border(
                border = BorderStroke(1.dp, color.copy(alpha = 0.4f)),
                shape = RoundedCornerShape(8.dp)
            )
            .padding(12.dp),
        verticalAlignment = Alignment.Top
    ) {
        Icon(Icons.Default.Info, contentDescription = null, tint = color, modifier = Modifier.size(20.dp))
        Spacer(Modifier.width(8.dp))
        Text(text, color = color, fontSize = 13.sp, lineHeight = 19.sp)
    }
}
